IoTuring/Configurator/ConfiguratorIO.py

```diff
@@ -0,0 +1,144 @@
+        
+    
+    def getFilePath(self):
+        """ Returns the path to the configurations file. """
+        return os.path.join(self.getFolderPath(), CONFIGURATION_FILE_NAME)
+    
+    def createFolderPathIfDoesNotExist(self):      
+        """ Check if file exists, if not check if path exists: if not create both folder and file otherwise just the file """
+        if not os.path.exists(self.getFolderPath()):    
+            if not os.path.exists(self.getFolderPath()):
+                os.makedirs(self.getFolderPath())        
+    
+    def getFolderPath(self):
+        """ Returns the path to the configurations file. If the directory where the file
+            will be stored doesn't exist, it will be created. """
+            
+        folderPath = self.defaultFolderPath()
+        try:
+            # Use path from environment variable if present, otherwise os specific folders, otherwise use default path
+            envvarPath = self.envvarFolderPath()
+            if envvarPath is not None:
+                folderPath = envvarPath
+            else:
+                _os = platform.system()
+                if _os == 'Darwin' and macos_support:
+                    folderPath = self.macOSFolderPath()
+                    folderPath = os.path.join(folderPath, self.directoryName)
+                elif _os == "Windows":
+                    folderPath = self.windowsFolderPath()        
+                    folderPath = os.path.join(folderPath, self.directoryName)
+                elif _os == "Linux":
+                    folderPath = self.linuxFolderPath()
+                    folderPath = os.path.join(folderPath, self.directoryName)
+        except:
+            pass # default folder path will be used
+        
+        # add slash if missing (for log reasons)
+        if not folderPath.endswith(os.sep):
+            folderPath += os.sep
+            
+        return folderPath
+
+    def defaultFolderPath(self):
+        return os.path.dirname(inspect.getfile(ConfiguratorIO))
+    
+    # https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/MacOSXDirectories/MacOSXDirectories.html
+    def macOSFolderPath(self):
+        paths = NSSearchPathForDirectoriesInDomains(NSApplicationSupportDirectory,NSUserDomainMask,True)
+        basePath = (len(paths) > 0 and paths[0]) or NSTemporaryDirectory()
+        return basePath
+    
+    # https://docs.microsoft.com/en-us/windows/win32/shell/knownfolderid
+    def windowsFolderPath(self):
+        return os.environ["APPDATA"]
+    
+    # https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
+    def linuxFolderPath(self):
+        return os.environ["XDG_CONFIG_HOME"] if "XDG_CONFIG_HOME" in os.environ else os.path.join(os.environ["HOME"], ".config")
+    
+    def envvarFolderPath(self):
+        return os.environ[CONFIG_PATH_ENV_VAR] if CONFIG_PATH_ENV_VAR in os.environ else None
```

Comment:
`os.getenv` returns None if the envvar doesn't exist:

```suggestion
        return os.getenv(CONFIG_PATH_ENV_VAR)
```